ext/reflection/php_reflection.c

```diff
@@ -3519,38 +3517,20 @@ ZEND_METHOD(ReflectionMethod, setAccessible)
 /* {{{ Constructor. Throws an Exception in case the given class constant does not exist */
 ZEND_METHOD(ReflectionClassConstant, __construct)
 {
-	zval *classname, *object;
+	zval *object;
 	zend_string *constname;
 	reflection_object *intern;
 	zend_class_entry *ce;
 	zend_class_constant *constant = NULL;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "zS", &classname, &constname) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_CLASS_NAME_OR_OBJ(ce)
```

Comment:
Unfortunately, I've just realized that using this macro will not be right because it will result in a `TypeError` instead of a `ReflectionException` if an invalid class name is given, and I think the latter is more appropriate in this case.

However, I believe we should throw a `TypeError` if any type besides `string|object` is provided, so that using `Z_PARAM_STR_OR_OBJ` ZPP is my preference.